jersey-servers/src/main/java/com/palantir/remoting2/servers/jersey/TraceEnrichingFilter.java

```diff
@@ -91,4 +98,24 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
             return Optional.of(header.equals("1"));
         }
     }
+
+    private static void addUserIdMetadataToSpan(ContainerRequestContext requestContext) {
+        Optional<String> userId = getUserIdFromAuthHeader(requestContext);
+        if (userId.isPresent()) {
+            Tracer.addSpanMetadata(ImmutableMap.of(USER_ID_METADATA_KEY, userId.get()));
+        }
+    }
+
+    private static Optional<String> getUserIdFromAuthHeader(ContainerRequestContext requestContext) {
```

Comment:
@markelliot - could you point me to where else we are doing this? I don't see auth-header related filters in http-remoting. Do you mean we already do this in the filters added from the multipass dropwizard bundle (e.g. `UserIdFilter`)?